### PR TITLE
fake_quant: fix device affinity and buffer resizing for state_dict

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50868 fake_quant: fix device affinity and buffer resizing for state_dict**

Summary:

Ensures that `FakeQuantize` respects device affinity when loading from
state_dict, and knows how to resize scale and zero_point values
(which is necessary for FQ classes wrapping per channel observers).

This is same as https://github.com/pytorch/pytorch/pull/44537, but for
`FakeQuantize`.

Test Plan:

```
python test/test_quantization.py TestObserver.test_state_dict_respects_device_affinity
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25991570](https://our.internmc.facebook.com/intern/diff/D25991570)

### CODE REVIEW

File: torch/quantization/fake_quantize.py [hunk 1]
```diff
@@ -170,7 +170,23 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
             key = prefix + name
             if key in state_dict:
                 val = state_dict[key]
-                setattr(self, name, val)
+                # Custom handling to allow loading min_vals or max_vals
```
Comment (jerryzh168) [inline torch/quantization/fake_quantize.py:173]: nit: loading scale and zero_point?

File: torch/quantization/fake_quantize.py [hunk 2]
```diff
@@ -170,7 +170,23 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
...
+                # buffers are resized here, and the values are copied in
+                # the default state_dict loading code of the parent.
+                if name == 'scale':
+                    self.scale.resize_(val.shape)
+                else:
+                    assert name == 'zero_point'
+                    self.zero_point.resize_(val.shape)
+                # For torchscript module we need to update the attributes here since we do not
+                # call the `_load_from_state_dict` function defined module.py
+                if torch.jit.is_scripting():
```
Comment (jerryzh168) [inline torch/quantization/fake_quantize.py:184]: looks like https://github.com/pytorch/pytorch/pull/44537/files does not have this code, should we add it there as well

Comment (vkuzo) [reply torch/quantization/fake_quantize.py:184]: yes, it was added later to that code path as well: https://github.com/pytorch/pytorch/blob/master/torch/quantization/observer.py#L590

**Review by jerryzh168 — APPROVED** [review]
LG

**Pull request closed without merging**